### DISCUSSION TITLE
[AngularFireAuthGuard] Expose ways to call canActivateChild on the guard or in the route definition

### DISCUSSION THREAD

**Discussion by Jak3b0** [angular/angularfire#2186] [Ideas]
### Version info

**Angular:** 9.0.0-next.8

**Firebase:** 6.6.1

**AngularFire:** 5.2.1

**Other (e.g. Ionic/Cordova, Node, browser, operating system):**

### Expected behavior

I need to be able to tell the guard to also check for child routes. This could be done via some sort of option I can pass to the guard or via using a `canActivateChild` exported by @angular/fire/auth-guard (like the `canActivate` one)

**Comment by debuggerpk** [comment]
is it okay if i submit a PR for this?

**Comment by playgamemy** [comment]
Is this function going to be added? I need that to be able to use that for canActivateChild method, but at the moment somehow it returns an error but not if I pass the guard to canActivate
